src/app/services/polling.service.ts

```diff
@@ -0,0 +1,66 @@
+import { Injectable } from '@angular/core';
+
+import {
+  BehaviorSubject,
+  delay,
+  from,
+  fromEvent,
+  mergeMap,
+  of,
+  repeatWhen,
+  share,
+  switchMap,
+  takeWhile,
+  timer,
+} from 'rxjs';
+
+import { OnlineStatusService, OnlineStatusType } from 'ngx-online-status';
```

Comment:
I guess we can get rid of any lib as we can have info we need from _document_ state